internal/bminventory/inventory.go

```diff
@@ -2135,6 +2135,23 @@ func setCommonUserNetworkManagedParams(params *models.ClusterUpdateParams, singl
 	return nil, false
 }
 
+func (b *bareMetalInventory) updatePlatformSources(params installer.UpdateClusterParams, cluster *common.Cluster, updates map[string]interface{}, usages map[string]models.Usage) {
+	b.updatePlatformParams(params, cluster, updates)
+
+	if params.ClusterUpdateParams.Platform.Type != models.PlatformTypeBaremetal {
```

Comment:
we should update the platform type not only for vsphere so this condition is not necessary.
we should always call b.setUsage(true, usage.PlatformSelectionUsage, &map[string]interface{}{
			"platform_type": params.ClusterUpdateParams.Platform.Type, "with_credentials": withCredentials}, usages)

Reply:
If nothing is changed why should we call it? We want to monitor the usage of this feature. If platform type is Baremetal you define it as feature usage?

Reply:
I agree with @eliorerz the usage is when we define a non bare metal platform. In the additional widget we will also break it up to the platform used and the relative percentage of the different platforms.

Reply:
we use the feature usage in our data dashboard and we would like to see in the future - how many clusters were with baremetal, vsphere, libvirt ... it will make our data analysis much easier

Reply:
We can determine how clusters are with baremetal using the total clusters about minus clusters with platform set

Reply:
the decision was to report only on changes and not the default